### PR TITLE
Fix blackbody_nu to accept arrays of temperatures

### PR DESCRIPTION
**Pull request by astrofrog**
I think this should be considered a bug fix to be included in 1.1 and backported to 1.0.x

@embray - so just to be clear, I milestone this as 1.0.x?

cc @pllim

### CODE REVIEW

File: astropy/analytic_functions/tests/test_blackbody.py [hunk 1]
```diff
@@ -105,3 +106,14 @@ def test_blackbody_exceptions_and_warnings():
         blackbody_nu(-1. * u.AA, 5000)
     assert len(w) == 1
     assert 'invalid' in w[0].message.args[0]
+
+
+def test_blackbody_array_temperature():
+
+    # Regression test to make sure that the temperature can be an array
+
+    flux = blackbody_nu(1.2 * u.mm, [100, 200, 300])
```
Comment (pllim) [inline astropy/analytic_functions/tests/test_blackbody.py]: I think a better test would be an array of wavelengths *and* an array of temperatures. It is uncommon for people to only evaluate blackbody model at a single wavelength value.

Comment (astrofrog) [inline astropy/analytic_functions/tests/test_blackbody.py]: Added - I left the single wavelength/multiple temperature case too

**Comment by pllim** [comment]
Also, I think this is more of a "new feature" rather than "bug fix". It was not intended to be used with multiple temperature values at once. I documented temperature as a singular form in the docstring, which should also be updated in your PR.

**Comment by astrofrog** [comment]
Ok, in that case I will remove the milestone so that it goes in 1.2

**Comment by pllim** [comment]
To revisit the broadcasting rule, what you have is fine, but I think is inconsistent with `astropy.modeling`. Feel free to correct me, @embray, as I realize that it has also undergone a lot of changes recently.

**Comment by astrofrog** [comment]
I'm flexible on the broadcasting, but what I have in this PR is what I personally think makes the most sense.

**Comment by pllim** [comment]
@astrofrog , okay. We'll worry about consistency with `modeling` when it supports `Quantity` and we have to move this over there.

**Comment by mhvk** [comment]
Looks good to me. Agreed that it is not really a bug fix, and also agreed that following numpy broadcasting rules is least likely to lead to confusion -- indeed, isn't it great that to get this one only had to change the tests?

**Comment by embray** [comment]
This can go in 1.0.6.  I think I ran into this when trying to work on implementing these as a model.  I think it's a bug because there's no reason this shouldn't have been supported before.

> Feel free to correct me, @embray, as I realize that it has also undergone a lot of changes recently.

Not really.  People keep assuming that but the broadcasting rules for models have been the same for like a year.  It should definitely follow the normal Numpy broadcasting rules.

**Comment by pllim** [comment]
I'm fine with the broadcasting clarified. Thanks.

As for bug or not bug, it was not *intended* to support array of temperature, regardless or not whether it *should* have. So, please consider that and label it however you wish.

I like this additional functionality, so thanks, @astrofrog :+1:

**Comment by mhvk** [comment]
If we call this a bug, then it should be listed in the bug fixes section. Other than that, I think this is ready to go in.

**Comment by pllim** [comment]
Okay, according to the chart below, I guess it is technically a "bug".

**Comment by embray** [comment]
I tweaked the changelog message and then merged in 68824c3aae72bcf77abee23a1c093b6758c9064c

(also backported to relevant release branches)

**Pull request closed without merging**